src/Sentry.AzureFunctions.Worker/SentryFunctionsWorkerMiddleware.cs

```diff
@@ -0,0 +1,82 @@
+using Microsoft.Azure.Functions.Worker;
+using Microsoft.Azure.Functions.Worker.Middleware;
+
+namespace Sentry.AzureFunctions.Worker;
+
+internal class SentryFunctionsWorkerMiddleware : IFunctionsWorkerMiddleware
+{
+    private readonly IHub _hub;
+
+    public SentryFunctionsWorkerMiddleware(IHub hub)
+    {
+        _hub = hub;
+    }
+
+    public async Task Invoke(FunctionContext context, FunctionExecutionDelegate next)
+    {
+        try
+        {
+            var transactionName = await GetTransactionNameAsync(context) ?? context.FunctionDefinition.Name;
+
+            _hub.ConfigureScope(scope =>
+            {
+                var transaction = _hub.StartTransaction(transactionName, "function");
```

Comment:
This starts the transaction, but we need to finish it later, and set status correctly.  See `src/Sentry.AspNetCore/SentryTracingMiddleware.cs` and follow a similar process.

Reply:
Ah, I missed the `transaction.Finish()` call. @mattjohnsonpint, if I get it right, a transaction should be finished regardless of the outcome. And if it's a failure, the transaction should be finished with the exception that was thrown. Did I get that part right?

Reply:
Also, unlike ASP.NET Core, Functions middleware will handle multiple types of triggers. Should it still handle the [span status](https://github.com/getsentry/sentry-dotnet/blob/0110385baedc3527d03652b4b6a9dbd8fe459029/src/Sentry.AspNetCore/SentryTracingMiddleware.cs#L200) or not?

Reply:
Yes, to both.

For an HTTP trigger, we should use the same approach as ASP.NET Core - taking the status from the outgoing HTTP response code.

For other trigger types, we can use `transaction.Finish()` on success, or `transaction.Finish(exception)` on failure.